metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,26 +139,75 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
+        elif raw_field.get("type") == "struct":
+            for field in raw_field.get("fields"):
+                if isinstance(field.get("type"), dict):
+                    parsed_struct += (
+                        "{0}:".format(field.get("name"))
+                        if parsed_struct
+                        else "" + "{0}:".format(field.get("name"))
+                    )
+                    parsed_struct += field.get("type").get("type") + "<"
+                    return self._inner_field_details(field.get("type"), parsed_struct)
+                else:
+                    parsed_struct = (
+                            parsed_struct
+                            + "{0}:{1}".format(field.get("name"), field.get("type"))
+                            + ","
+                    )
+
+            parsed_struct = parsed_struct.rstrip(",")
+            parsed_struct = (
+                parsed_struct + ">"
+                if parsed_struct.count(">") < parsed_struct.count("<")
+                else parsed_struct
+            )
+            return parsed_struct
+        else:
+            return str(raw_field.get("elementType"))
+
+    def _parse_datatype(self, raw_field_json_str: str) -> List[SchemaFieldClass]:
+        raw_field_json = json.loads(raw_field_json_str)
+
+        # get the parent field name and type
+        field_name, field_type = _get_parent_field_details(raw_field_json)
+
+        inner_field = raw_field_json.get("type")
+
+        if isinstance(inner_field, str):
+            return get_schema_fields_for_hive_column(field_name, field_type)
+        else:
+            # recursively parse the fields to get the native data type
+            inner_data_type = self._inner_field_details(inner_field)
+
+            native_data_type = field_type + "<" + inner_data_type + ">"
+
+            avro_schema_data = get_schema_fields_for_hive_column(
+                field_name, native_data_type
+            )
+            return avro_schema_data
+
     def get_fields(self, delta_table: DeltaTable) -> List[SchemaField]:
         fields: List[SchemaField] = []
 
         for raw_field in delta_table.schema().fields:
-            field = SchemaField(
-                fieldPath=raw_field.name,
-                type=SchemaFieldDataType(
-                    tableschema_type_map.get(raw_field.type.type, NullTypeClass)()
-                ),
-                nativeDataType=raw_field.type.type,
-                recursive=False,
-                nullable=raw_field.nullable,
-                description=str(raw_field.metadata),
-                isPartitioningKey=True
-                if raw_field.name in delta_table.metadata().partition_columns
-                else False,
-            )
-            fields.append(field)
-        fields = sorted(fields, key=lambda f: f.fieldPath)
+            field = self._parse_datatype(raw_field.to_json())
+            fields = fields + field
 
+        fields = sorted(fields, key=lambda f: f.fieldPath)
```

Comment:
any particular reasons to add this sorting ?

Reply:
it was already there in the earlier codebase, so have kept it

Reply:
in that case, you can ignore this comment,